### ISSUE TITLE
Redis hScan does not return a hash

### ISSUE THREAD

**Issue by jimyaghi** [facebook/hhvm#7294]
### HHVM Version 
3.15.0-dev

### Standalone code, or other way to reproduce the problem
while( ($paths = $this->redis->hScan("paths", $cursor, $pattern)) !== false ){
            print_r($paths);
}

### Expected result
Array
(
   [/1/568128/] => 3
)
### Actual result
Array
(
    [0] => /1/568128/
    [1] => 3
)

The problem is that redis returns a multi-line response for hScan, with odd numbered lines representing the member key and even numbered lines representing the value, just like with zScan. 

See http://redis.io/commands/scan

Return value
SCAN, SSCAN, HSCAN and ZSCAN return a two elements multi-bulk reply, where the first element is a string representing an unsigned 64 bit number (the cursor), and the second element is a multi-bulk with an array of elements.
SCAN array of elements is a list of keys.
SSCAN array of elements is a list of Set members.
HSCAN array of elements contain two elements, a field and a value, for every returned element of the Hash.
ZSCAN array of elements contain two elements, a member and its associated score, for every returned element of the sorted set.

To summarise, SSCAN and SCAN have similar responses, while HSCAN and ZSCAN have similar responses. 

~jim

**Comment by aorenste** [comment]
I think this was fixed in #7334
